Add Cmake cross-compilation target for Mingw/Windows

scrubbbbs opened the pull request:
Adds necessary cmake files to compile the Windows version on a Linux/ Unix host. Also includes a binary packaging script to make a Windows portable package.

This is intended for MXE as the cross compiling environment, since its the easiest method. However, there isn't much specific to MXE so it could support other methods, or maybe native Mingw on Windows.

The build recipe is roughly  (untested; based  on  my bash history):

```shell
apt install wine

cd ~/
git clone <nomacs url>
git clone <mxe url>
cd mxe

# note: mxe may have apt packages to sidestep compiling things
# note: this is for qt5
make MXE_TARGETS=`x86_64-w64-mingw32.shared` qtbase qtimageformats qtwinextras opencv quazip tiff exiv2 libraw

export MXE_DIR=~/mxe
export MXE_TARGET=x86_64-w64-mingw32.shared
export PATH="${MXE_DIR}/usr/bin:$PATH"

alias pkg-config=${MXE_TARGET}-pkg-config
alias qmake=${MXE_TARGET}-qmake
alias cmake=${MXE_TARGET}-cmake

cd ~/
mkdir build-nomacs
cd build-nomacs

cmake -D QT_VERSION_MAJOR=5 -D ENABLE_TRANSLATIONS=ON -D ENABLE_HEIF=ON -D ENABLE_AVIF=ON  \
-DENABLE_JXL=ON -D USE_SYSTEM_QUAZIP=OFF -D CMAKE_BUILD_TYPE=Debug ~/nomacs/ImageLounge

make -j8
make collect # build a runnable package

wine nomacs-mingw/nomacs.exe # run nomacs under Wine (actually works OK, but slower)
```

novomesk commented:
Can try to investigate/check how this impact those who would like to build nomacs under MSYS2 ?

If we can support the MSYS2, that would be great.

Take a look at https://github.com/nomacs/nomacs/issues/853 , there is some patch there.

scrubbbbs commented:
Moved some stuff around and added Msys2 support for Qt  5, easy to make a running build but will need some tweaks for packaging; mainly another strategy for pulling all the dependent dlls.

I also fixed a problem with configuring the Unix/Linux build with Quazip-1 (was missing WITH_QUAZIP define). And I replaced tabs with spaces in that file as well (very annoying!).

Note that Qt 6 is not working on either target (mxe or msys2/mingw), (not a priority right now?) but should be straightforward to add.

The recipe for Msys2/mingw
```shell
export target=mingw-w64-x86_64

pacman -S $target-qt5-base $target-qt5-svg $target-qt5-winextras $target-qt5-tools \
          $target-libraw $target-libtiff $target-exiv2 $target-opencv $target-cmake $target-gcc

export PATH=/ming64:$PATH

cd <build-dir>
cmake  -D  QT_MAJOR_VERSION=5  -D ENABLE_TRANSLATIONS=ON -D  USE_SYSTEM_QUAZIP=ON <nomacs-dir>/ImageLounge

cmake --build . --parallel 8
```
